packages/dashboard/src/helpers/updateChildFilters.ts

```diff
@@ -0,0 +1,50 @@
+import { SharedFilter } from '../types/SharedFilter'
+import _ from 'lodash'
+
+export const updateChildFilters = (newSharedFilters: SharedFilter[], data: Record<string, any>): SharedFilter[] => {
+  const dataSet = Object.values(data).flat()
+
+  // Find indexes of all child filters
+  const childFilterIndexes: number[] = newSharedFilters
+    .map((filter, index) => (filter.type === 'datafilter' && filter.parents ? index : -1))
+    .filter(index => index !== -1)
+  if (childFilterIndexes.length === 0) return newSharedFilters
+
+  // deep copy of the shared filters
+  const updatedFilters = _.cloneDeep(newSharedFilters)
+
+  // Update each child filter
+  childFilterIndexes.forEach(childIndex => {
+    const childFilter: SharedFilter = newSharedFilters[childIndex]
+    const parentFilter: SharedFilter = newSharedFilters.find(
+      filter => String(childFilter.parents) === String(filter.key)
```

Comment:
childFilter.parents is an Array<string>

Reply:
I think it is strings. A child filter can not have multiple parents from what we have.